src/rlookup.c

```diff
@@ -538,8 +538,8 @@ static int RLookup_HGETALL(RLookup *it, RLookupRow *dst, RLookupLoadOptions *opt
   // and when the deployment is not enterprise-crdt
   if(!isFeatureSupported(RM_SCAN_KEY_API_FIX) || isCrdt){
     rep = RedisModule_Call(ctx, "HGETALL", "s", krstr);
-
     if (rep == NULL || RedisModule_CallReplyType(rep) != REDISMODULE_REPLY_ARRAY) {
+      QueryError_SetCode(options->status, QUERY_ENODOC);
```

Comment:
Consider removing this and count on the return value

Reply:
Using now the return value
This is inline with other load functions such as `getKeyCommonHash`

Reply:
So why you still need to set the error code here? I do not mind rearrange stuff but lets do it on separate PR, remember that this PR need to be cherrypicked.